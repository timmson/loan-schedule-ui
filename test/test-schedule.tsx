import React from "react"
import renderer, {act} from "react-test-renderer"
import Context from "../src/context"

import Schedule from "../src/schedule"

describe("Schedule should", () => {

	let component = null
	const dispatch = () => {}
	const schedule = {
		payments: [
			{
				principalAmount: "1",
				interestAmount: "0"
			}
		]
	}

	const createComponent = (schedule, dispatch) => renderer.create(
		<Context.Provider value={dispatch}>
			<Schedule schedule={schedule}/>
		</Context.Provider>
	)

	test("trigger click", () => {
		component = createComponent(schedule, dispatch)

		act(() =>
			component.root.findByType("tbody").children[0].props.onClick()
		)

		//component.toJSON()
	})

	test("equal to snapshot", () => {
		component = createComponent(schedule, dispatch)

		expect(component.toJSON()).toMatchSnapshot()
	})

})