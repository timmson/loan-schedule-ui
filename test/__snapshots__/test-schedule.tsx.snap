// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule should equal to snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-12 text-center"
  >
    <table
      className="table table-hover table-bordered w-100 border-dark"
    >
      <thead
        className="table-secondary"
      >
        <tr
          className="desktop"
        >
          <th
            rowSpan={2}
          >
            Дата
          </th>
          <th
            rowSpan={2}
          >
            Платеж, ₽
          </th>
          <th
            rowSpan={2}
          >
            Мин. платеж, ₽
          </th>
          <th
            colSpan={2}
          >
            включая
          </th>
          <th
            rowSpan={2}
          >
            Остаток, ₽
          </th>
        </tr>
        <tr
          className="desktop"
        >
          <th>
            Тело долга, $
          </th>
          <th>
            Проценты, $
          </th>
        </tr>
        <tr
          className="mobile"
        >
          <th>
            Дата
          </th>
          <th>
            Платеж, $
          </th>
          <th>
            Проценты, $
          </th>
          <th>
            Остаток, $
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="orange"
          onClick={[Function]}
          style={
            {
              "cursor": "pointer",
            }
          }
        >
          <td />
          <td />
          <td
            className="desktop-cell"
          />
          <td
            className="desktop-cell"
          >
            1.00
          </td>
          <td>
            0.00
          </td>
          <td />
        </tr>
      </tbody>
      <tfoot>
        <tr>
          <td>
             
          </td>
          <td />
          <td
            className="desktop-cell"
          >
             
          </td>
          <td
            className="desktop-cell"
          />
          <td />
          <td>
             
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;
